src/libs/dutil/WixToolset.DUtil/butil.cpp

```diff
@@ -153,113 +162,26 @@ DAPI_(HRESULT) BundleEnumRelatedBundle(
-        
-        hr = RegOpen(hkRoot, sczUninstallSubKeyPath, KEY_READ, &hkBundle);
-        ButilExitOnFailure(hr, "Failed to open uninstall key path.");
-
-        // If it's a bundle, it should have a BundleUpgradeCode value of type REG_SZ (old) or REG_MULTI_SZ
-        hr = RegGetType(hkBundle, BUNDLE_REGISTRATION_REGISTRY_BUNDLE_UPGRADE_CODE, &dwType);
-        if (FAILED(hr))
-        {
-            ReleaseRegKey(hkBundle);
-            ReleaseNullStr(sczUninstallSubKey);
-            ReleaseNullStr(sczUninstallSubKeyPath);
-            // Not a bundle
-            continue;
-        }
-
-        switch (dwType)
-        {
-            case REG_SZ:
-                hr = RegReadString(hkBundle, BUNDLE_REGISTRATION_REGISTRY_BUNDLE_UPGRADE_CODE, &sczValue);
-                ButilExitOnFailure(hr, "Failed to read BundleUpgradeCode string property.");
-
-                if (CSTR_EQUAL == ::CompareStringW(LOCALE_INVARIANT, NORM_IGNORECASE, sczValue, -1, wzUpgradeCode, -1))
-                {
-                    *pdwStartIndex = dwIndex;
-                    fUpgradeCodeFound = TRUE;
-                    break;
-                }
-
-                ReleaseNullStr(sczValue);
-
-                break;
-            case REG_MULTI_SZ:
-                hr = RegReadStringArray(hkBundle, BUNDLE_REGISTRATION_REGISTRY_BUNDLE_UPGRADE_CODE, &rgsczBundleUpgradeCodes, &cBundleUpgradeCodes);
-                ButilExitOnFailure(hr, "Failed to read BundleUpgradeCode  multi-string property.");
-
-                for (DWORD i = 0; i < cBundleUpgradeCodes; i++)
-                {
-                    LPWSTR wzBundleUpgradeCode = rgsczBundleUpgradeCodes[i];
-                    if (wzBundleUpgradeCode && *wzBundleUpgradeCode)
-                    {
-                        if (CSTR_EQUAL == ::CompareStringW(LOCALE_INVARIANT, NORM_IGNORECASE, wzBundleUpgradeCode, -1, wzUpgradeCode, -1))
-                        {
-                            *pdwStartIndex = dwIndex;
-                            fUpgradeCodeFound = TRUE;
-                            break;
-                        }
-                    }
-                }
-                ReleaseNullStrArray(rgsczBundleUpgradeCodes, cBundleUpgradeCodes);
-
-                break;
-
-            default:
-                ButilExitWithRootFailure(hr, E_NOTIMPL, "BundleUpgradeCode of type 0x%x not implemented.", dwType);
-        }
-
-        if (fUpgradeCodeFound)
-        {
-            if (psczBundleId)
-            {
-                *psczBundleId = sczUninstallSubKey;
-                sczUninstallSubKey = NULL;
-            }
-
-            break;
-        }
-
-        // Cleanup before next iteration
-        ReleaseRegKey(hkBundle);
-        ReleaseNullStr(sczUninstallSubKey);
-        ReleaseNullStr(sczUninstallSubKeyPath);
+#if defined(_WIN64)
+        hr = DoBundleEnumRelatedBundle(hkRoot, REG_KEY_32BIT, wzUpgradeCode, pdwStartIndex, psczBundleId);
```

Comment:
`pdwStartIndex` was a shortcut for a context parameter but now we have to keep track of the current bitness, too. This function will skip the first `*pdwStartIndex` entries here.